sdk/highlight-ruby/e2e/rails/blog/app/controllers/application_controller.rb

```diff
@@ -0,0 +1,13 @@
+require 'highlight'
+
+class ApplicationController < ActionController::Base
+    include Highlight::Integrations::Rails
+
+    around_action :highlight_wrapper
```

Comment:
@et any ideas to simplify this? not sure if it's possible, but I'd like this to be `around_action Highlight::Integrations::Rails::with_highlight_context`, issue I was running into is that I need to call `request` from this method. Is there a way for code within the `with_highlight_context` method to call a method of the including class? `ApplicationController` in this case

Reply:
for context, this is an example of how a Rails app would add error reporting

Reply:
I don't think it's possible because [request](https://api.rubyonrails.org/classes/ActionController/Base.html#method-i-request) is a method on an instance of application controller, not the class itself.

It may be possible with some metaprogramming since what you have here is very readable.

Reply:
oh that's just the clue I needed, thanks. made `with_highlight_context` an instance method and this works fine and doesn't require the user to write the extra boilerplate